bigquery/storage_iterator.go

```diff
@@ -191,20 +201,6 @@ func (it *arrowIterator) init() error {
 		return iterator.Done
 	}
 
-	if it.schema == nil {
-		meta, err := it.session.table.Metadata(it.ctx)
-		if err != nil {
-			return err
-		}
-		it.schema = meta.Schema
-	}
-
-	decoder, err := newArrowDecoderFromSession(it.session, it.schema)
-	if err != nil {
-		return err
-	}
-	it.decoder = decoder
-
 	wg := sync.WaitGroup{}
 	wg.Add(len(streams))
 	sem := semaphore.NewWeighted(int64(it.session.settings.maxWorkerCount))
```

Comment:
would it be feasible to expose the streams themselves to a consumer to allow them to control the parallelization instead of forcing them to a specific route like this?